wordpress-mobile/WordPress-iOS
# Offline Mode: Remove Deprecated Code – P9

- Remove unused `isFeaturedImageChanged` – starting with 24.9 the app automatically diffs the changes and sends only the fields with changes to the server (aka "partial updates"), so there is no need to manually decide when to send the featured image ID.
- Update `PreviewNonceHandler` to use new `post.changes` API instead of `hasLocalChanges` – this is something that is now used everywhere and is based on the same diffing algorithm; we just forgot to update `PreviewNonceHandler` in the previous release.
- Remove unused `PostHelper.remotePostWithPost` (https://github.com/wordpress-mobile/WordPress-iOS/pull/23321 and other PRs were prerequisites for this change)
- Simplify "Duplicate" feature by removing the need to apply the autosave – there is absolutely no reason to do it. Autosaves are handled inline in the editor now, and this is just something we missed in 24.9.
- Remove a bunch of other now unused code

## To test:

- Re-run the tests from https://github.com/wordpress-mobile/WordPress-iOS/pull/23308
- ✅  Verify that "Duplicate" button creates a copy of the post and opens it in the editor

## Regression Notes
1. Potential unintended areas of impact: Post List
2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
